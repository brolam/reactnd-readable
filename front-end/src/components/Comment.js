import React from 'react'
import PropTypes from 'prop-types'
import Moment from 'moment';
import VoteScore from './VoteScore'

function Comment(props) {
  const { comment } = props
  return (
    <div className="post-comment" >
      <div className="post-comment-author">
        by {comment.author} {Moment(comment.timestamp).from(new Date())}
      </div>
      <div>{comment.body}</div>
      <div className="post-comment-footer">
        <button className="edit-button" href="/" onClick={e => { }}>Edit</button>
        <button className="delete-button" onClick={e => { }}>Delete</button>
        <VoteScore voteScore={comment.voteScore}
        />
      </div>
    </div>
  )
}

Comment.propTypes = {
  comment: PropTypes.shape({
    id: PropTypes.string.isRequired,
    timestamp: PropTypes.number.isRequired,
    body: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    voteScore: PropTypes.number.isRequired,
  })
}

export default Comment