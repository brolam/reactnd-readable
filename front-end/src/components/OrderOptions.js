import React from 'react'

function OrderOptions(props) {
  return (
    <div className="order-options">
      <select>
        <option value="voteScore" >Vote Score</option>
        <option value="publishedDate" >Published date</option>
      </select>
    </div>
  )
}

export default OrderOptions