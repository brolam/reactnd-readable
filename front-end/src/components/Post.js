import React from 'react'
import PropTypes from 'prop-types'
import Moment from 'moment';
import VoteScore from './VoteScore'

function Post(props) {
  const { post } = props
  return (
    <div className="post" onClick={props.onSelected}>
      <div className="post-header">
        <div className="post-header-title">
          <span className={"category " + post.category} >{post.category[0].toUpperCase() + post.category.slice(1)}</span> {post.title}
        </div>
        <div className="post-header-published">
          <span className="published">{Moment(post.timestamp).from(new Date())}</span>
        </div>
      </div>
      <div className="post-header-author">by {post.author}</div>
      <div className="post-body">{post.body}</div>
      <div className="post-footer">
        <VoteScore voteScore={post.voteScore} />
      </div>
    </div>
  )
}

Post.propTypes = {
  post: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    voteScore: PropTypes.number.isRequired,
  }),
  onSelected: PropTypes.func.isRequired
}

export default Post