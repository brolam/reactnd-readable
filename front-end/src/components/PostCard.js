import React from 'react'
import PropTypes from 'prop-types'
import Moment from 'moment';

function PostCard({ post, onClick }) {
  return (
    <div className="col-6 col-lg-4" onClick={onClick} >
      <div className={`card text-white bg-${getColorClass(post.category)} mb-3`} >
        <div className="card-header">{post.category}</div>
        <div className="card-body">
          <h4 className="card-title">{post.title}</h4>
          <p className="card-text">{post.body}</p>
          <div className="card-text"><small className="card-text">{post.author}, {Moment(post.timestamp).from(new Date())}.</small></div>
        </div>
      </div>
    </div>
  )
}

PostCard.propTypes = {
  post: PropTypes.shape({
    category: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    timestamp: PropTypes.number.isRequired,
    author: PropTypes.string.isRequired,
    voteScore: PropTypes.number.isRequired
  }),
  onClick: PropTypes.func.isRequired
}

export const getColorClass = (category) => {
  const categores = ['udacity', 'react', 'redux'];
  const colors = ['primary', 'info', 'danger'];
  const indexColor = categores.indexOf(category);
  return indexColor === -1 ? 'dark' : colors[indexColor];
}

export default PostCard