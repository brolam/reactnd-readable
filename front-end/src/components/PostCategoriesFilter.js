import React from 'react'
import PropTypes from 'prop-types'

function PostCategoriesFilter(
  { categories,
    onSelected = e => { },
    selectedCategoryPath = 'none' }) {
  return (
    <select
      value={selectedCategoryPath}
      onChange={onSelected} >
      <option value="none">All Categories</option>
      {categories.map(category => (
        <option
          key={category.name}
          value={category.path} >{category.name[0].toUpperCase() + category.name.slice(1)}</option>
      ))}
    </select>
  )
}

PostCategoriesFilter.propTypes = {
  categories: PropTypes.array.isRequired
}

export default PostCategoriesFilter