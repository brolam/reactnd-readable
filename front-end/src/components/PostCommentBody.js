import React from 'react'

function PostCommentBody({ post }) {
  return (<div></div>)
}

export default PostCommentBody