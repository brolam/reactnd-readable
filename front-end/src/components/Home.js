import React from 'react'
import PostCategoriesFilter from '../components/PostCategoriesFilter'
import SearchBar from '../components/SearchBar'
import PostList from '../components/PostList'
import PostModal from '../components/PostModal'
import WaitProcessModal from '../components/WaitProcessModal'

function Home({ posts, categories, isNewPost, isShowWaitProcessModal }) {
  return (
    <div className="app">
      <div className="main-page-header">
        <div className="main-page-header-title">
          <span />
          <h1>Readable</h1>
          <PostCategoriesFilter categories={categories} onSelected={e => { }} />
        </div>
      </div>
      <SearchBar placeholder="Search by title post" />
      <div className="main-page-content">
        <PostList posts={posts} />
      </div>
      {
        isNewPost && (
          <PostModal
            post={{}}
            categories={categories}
            onClickBackButton={e => { }}
            onSave={e => { }}
          />
        )
      }
      {
        isShowWaitProcessModal && (
          <WaitProcessModal
            message="Please wait while the information is updated."
          />
        )
      }
    </div>
  )
}

export default Home