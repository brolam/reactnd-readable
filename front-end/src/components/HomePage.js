import React from 'react'
import PropTypes from 'prop-types'
import PostCategoriesFilter from '../components/PostCategoriesFilter'
import SearchBar from '../components/SearchBar'
import PostList from '../components/PostList'
import PostModal from '../components/PostModal'
import WaitProcessModal from '../components/WaitProcessModal'

function HomePage({
  posts,
  categories,
  isNewPost = false,
  isShowWaitProcessModal = false,
  goPostNew = e => { },
  goHome = e => { },
  goHomeFilterByCategory = (categoryPath) => { },
  onSavePost = fieldsWasValidated => { },
  onSelectedPost = (post) => { },
  onVoteScorePost = (postId, option) => { },
  onChangeOrderPostsList = (order) => { } }) {
  return (
    <div className="app">
      <div className="main-page-header">
        <div className="main-page-header-title">
          <span />
          <h1>Readable</h1>
          <PostCategoriesFilter
            categories={categories}
            onSelected={e => { goHomeFilterByCategory(e.target.value) }} />
        </div>
      </div>
      <SearchBar placeholder="Search by title post" />
      <div className="main-page-content">
        <PostList
          posts={posts}
          onSelectedPost={onSelectedPost}
          onVoteScorePost={onVoteScorePost}
          onChangeOrderPostsList={onChangeOrderPostsList} />
      </div>
      {
        isNewPost && (
          <PostModal
            post={{}}
            categories={categories}
            onClickBackButton={goHome}
            onSavePost={onSavePost}
          />
        )
      }
      {
        isShowWaitProcessModal && (
          <WaitProcessModal
            message="Please wait while the information is updated."
          />
        )
      }
      <div className="flat-button" onClick={goPostNew}>
        <a className="add">Add Post</a>
      </div>
    </div>
  )
}

HomePage.propTypes = {
  posts: PropTypes.array.isRequired,
  categories: PropTypes.array.isRequired,
}

export default HomePage