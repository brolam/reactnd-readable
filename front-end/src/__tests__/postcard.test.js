import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PostCard from '../components/PostCard';
import { getColorClass } from '../components/PostCard';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<PostCard post={postReact} onClick = {onClick} />, div);
});

test('last Snapshot category Udacity ', () => {
  const postCard = renderer.create(
    <PostCard post={postUdacity} onClick = {onClick} />
  );
  let tree = postCard.toJSON();
  expect(tree).toMatchSnapshot();
});

test('last Snapshot category React ', () => {
  const postCard = renderer.create(
    <PostCard post={postReact} onClick = {onClick} />
  );
  let tree = postCard.toJSON();
  expect(tree).toMatchSnapshot();
});

test('last Snapshot category Redux ', () => {
  const postCard = renderer.create(
    <PostCard post={postRedux} onClick = {onClick} />
  );
  let tree = postCard.toJSON();
  expect(tree).toMatchSnapshot();
});

test('last Snapshot wrong category', () => {
  const postCard = renderer.create(
    <PostCard post={postWrongCategory} onClick = {onClick} />
  );
  let tree = postCard.toJSON();
  expect(tree).toMatchSnapshot();
});

const postUdacity = {
  category: "udacity",
  id: "7ni6ok3ym7mf1p33lnez",
  title: "Udacity is the best place to learn technology.",
  timestamp: 1467166872634,
  author: "thingtwo",
  body: "Everyone says so after all.",
  deleted: false,
  voteScore: 5
}

const onClick = () =>{

} 

const postReact = {
  category: "react",
  id: "8xf0y6ziyjabvozdd253nd",
  title: "Udacity is the best place to learn React",
  timestamp: 1467166872634,
  author: "thingtwo",
  body: "Everyone says so after all.",
  deleted: false,
  voteScore: 5
}

const postRedux = {
  category: "redux",
  id: "7xf0y6ziyjabvozdd253nd",
  title: "Udacity is the best place to learn Redux",
  timestamp: 1467166872634,
  author: "thingtwo",
  body: "Everyone says so after all.",
  deleted: false,
  voteScore: 5
}

const postWrongCategory = {
  category: "xxxxxx",
  id: "7xf0y6ziyjabvozdd253nd",
  title: "Udacity is the best place to learn Redux",
  timestamp: 1467166872634,
  author: "thingtwo",
  body: "Everyone says so after all.",
  deleted: false,
  voteScore: 5
}
