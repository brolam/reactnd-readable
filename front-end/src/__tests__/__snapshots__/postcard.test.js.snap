// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot category React  1`] = `
<div
  className="col-6 col-lg-4"
  onClick={[Function]}
>
  <div
    className="card text-white bg-info mb-3"
  >
    <div
      className="card-header"
    >
      react
    </div>
    <div
      className="card-body"
    >
      <h4
        className="card-title"
      >
        Udacity is the best place to learn React
      </h4>
      <p
        className="card-text"
      >
        Everyone says so after all.
      </p>
      <div
        className="card-text"
      >
        <small
          className="card-text"
        >
          thingtwo
          , 
          a year ago
          .
        </small>
      </div>
    </div>
  </div>
</div>
`;

exports[`last Snapshot category Redux  1`] = `
<div
  className="col-6 col-lg-4"
  onClick={[Function]}
>
  <div
    className="card text-white bg-danger mb-3"
  >
    <div
      className="card-header"
    >
      redux
    </div>
    <div
      className="card-body"
    >
      <h4
        className="card-title"
      >
        Udacity is the best place to learn Redux
      </h4>
      <p
        className="card-text"
      >
        Everyone says so after all.
      </p>
      <div
        className="card-text"
      >
        <small
          className="card-text"
        >
          thingtwo
          , 
          a year ago
          .
        </small>
      </div>
    </div>
  </div>
</div>
`;

exports[`last Snapshot category Udacity  1`] = `
<div
  className="col-6 col-lg-4"
  onClick={[Function]}
>
  <div
    className="card text-white bg-primary mb-3"
  >
    <div
      className="card-header"
    >
      udacity
    </div>
    <div
      className="card-body"
    >
      <h4
        className="card-title"
      >
        Udacity is the best place to learn technology.
      </h4>
      <p
        className="card-text"
      >
        Everyone says so after all.
      </p>
      <div
        className="card-text"
      >
        <small
          className="card-text"
        >
          thingtwo
          , 
          a year ago
          .
        </small>
      </div>
    </div>
  </div>
</div>
`;

exports[`last Snapshot wrong category 1`] = `
<div
  className="col-6 col-lg-4"
  onClick={[Function]}
>
  <div
    className="card text-white bg-dark mb-3"
  >
    <div
      className="card-header"
    >
      xxxxxx
    </div>
    <div
      className="card-body"
    >
      <h4
        className="card-title"
      >
        Udacity is the best place to learn Redux
      </h4>
      <p
        className="card-text"
      >
        Everyone says so after all.
      </p>
      <div
        className="card-text"
      >
        <small
          className="card-text"
        >
          thingtwo
          , 
          a year ago
          .
        </small>
      </div>
    </div>
  </div>
</div>
`;
